packages/@aws-cdk-testing/framework-integ/test/aws-apigateway/test/integ.parametrized-swagger.api.ts

```diff
@@ -0,0 +1,58 @@
+import * as path from 'path';
+import * as lambda from 'aws-cdk-lib/aws-lambda';
+import * as cdk from 'aws-cdk-lib';
+import { IntegTest, ExpectedResult } from '@aws-cdk/integ-tests-alpha';
+import { Bucket } from 'aws-cdk-lib/aws-s3';
+import * as apigateway from 'aws-cdk-lib/aws-apigateway';
+import { DeployTimeSubstitutedFile } from 'aws-cdk-lib/aws-s3-deployment';
+
+let lambdaFunctionArn: string = '';
+class Test extends cdk.Stack {
+  constructor(scope: cdk.App, id: string) {
+    super(scope, id);
+
+    const hello = new lambda.Function(this, 'Hello', {
+      runtime: lambda.Runtime.NODEJS_14_X,
+      handler: 'index.handler',
+      code: lambda.Code.fromInline(`exports.handler = ${helloCode}`),
+    });
+
+    lambdaFunctionArn = hello.functionArn;
+
+    const bucket = new Bucket(this, 'Bucket');
+    const deployment = new DeployTimeSubstitutedFile(this, 'Deployment', {
+      source: path.join(__dirname, 'sample-definition-substitution.yaml'),
+      destinationBucket: bucket,
+      substitutions: {
+        testMethod: hello.functionArn,
+        xxxx: hello.functionArn,
+      },
+    });
+
+    new apigateway.SpecRestApi(this, 'sample-api-definition', {
+      apiDefinition: apigateway.ApiDefinition.fromBucket(deployment.bucket, deployment.objectKey),
+    });
+  }
+}
+
+const app = new cdk.App();
+new Test(app, 'my-stack');
+
+const testCase = new Test(app, 'test-apigateway-parameterized-swagger');
+const integ = new IntegTest(app, 'apigateway-parameterized-swagger', {
+  testCases: [testCase],
+});
+
+const invoke = integ.assertions.invokeFunction({
+  functionName: lambdaFunctionArn,
+});
+invoke.expect(ExpectedResult.objectLike({
```

Comment:
This doesn't prove anything relevant to this feature. The function could be deployed correctly, but the Rest API (which is the resource that depends on the substitutions being correct) could still be using an unsubstituted Open API file. At the very least, the assertion has to call the Rest API. You'll need to be creative and construct an API in such a way that its response contains the substituted variable. Response templates, perhaps?